src/elliptic/curves/secp256_k1.rs

```diff
@@ -174,25 +174,19 @@ impl ECScalar for Secp256k1Scalar {
     }
 
     fn from_bigint(n: &BigInt) -> Secp256k1Scalar {
+        let n = n.modulus(Self::group_order());
         if n.is_zero() {
             return Self::zero();
```

Comment:
My bad - purpose needs to be overriden
```suggestion
            return Secp256k1Scalar {
                purpose: "from_bigint",
                fe: Self::zero().fe,
            }
```